js/apps/admin-ui/src/realm-settings/RealmSettingsTabs.tsx

```diff
@@ -199,14 +199,15 @@ export const RealmSettingsTabs = ({
     }
 
     try {
-      await adminClient.realms.update(
-        { realm: realmName },
-        {
-          ...realm,
-          ...r,
-          id: r.realm,
-        }
-      );
+      const savedRealm = {
```

Comment:
I think we should add a type annotation here, the inferred type is very dynamic.

```suggestion
      const savedRealm: RealmRepresentation = {
```